browser/profiles/brave_bookmark_model_loaded_observer.cc

```diff
@@ -23,16 +26,19 @@ BraveBookmarkModelLoadedObserver::BraveBookmarkModelLoadedObserver(
 void BraveBookmarkModelLoadedObserver::BookmarkModelLoaded(
     BookmarkModel* model,
     bool ids_reassigned) {
+  if (!profile_->GetPrefs()->GetBoolean(kOtherBookmarksMigrated)) {
 #if BUILDFLAG(ENABLE_BRAVE_SYNC)
-  BraveProfileSyncServiceImpl* brave_profile_service =
+    BraveProfileSyncServiceImpl* brave_profile_service =
       static_cast<BraveProfileSyncServiceImpl*>(
           ProfileSyncServiceFactory::GetForProfile(profile_));
-  // When sync is enabled, we need to send migration records to other devices so
-  // it is handled in BraveProfileSyncServiceImpl::OnSyncReady
-  if (brave_profile_service && !brave_profile_service->IsBraveSyncEnabled())
-    BraveMigrateOtherNode(model);
+    // When sync is enabled, we need to send migration records to other devices
+    // so it is handled in BraveProfileSyncServiceImpl::OnSyncReady
+    if (brave_profile_service && !brave_profile_service->IsBraveSyncEnabled())
+      BraveMigrateOtherNodeFolder(model);
```

Comment:
nit: it's a little confusing to put this in the `#if` and `#else`

Reply:
This is for when sync is running, we don't migrate right away. It is guarded by `ENABLE_BRAVE_SYNC`, if it is false, we don't have access to `BraveProfileSyncServiceImpl`.
I don't see how confusing it is.